synthesizer/src/vm/finalize.rs

```diff
@@ -791,20 +792,68 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
+                            ensure!(
+                                committee.is_committee_member_open(*validator_address),
+                                "Ratify::Genesis(..) the delegator {address} is delegating to a closed validator {validator_address}",
+                            );
+                        }
+                        // Ensure that each staker has an entry in `public_balances`.
+                        ensure!(
+                            public_balances.contains_key(address),
+                            "Ratify::Genesis(..) the staker {address} is missing from the public balances",
+                        );
+                        // Accumulate the staked amount per validator.
+                        let total = stake_per_validator.entry(validator_address).or_insert(0u64);
+                        *total = total.saturating_add(*amount);
+                    }
+                    // Ensure the stake per validator matches the committee.
+                    ensure!(
+                        stake_per_validator.len() == committee.members().len(),
+                        "Ratify::Genesis(..) the number of validators in the committee does not match the number of validators in the bonded balances",
+                    );
+
+                    // Check that `committee` is consistent with `stake_per_validator`.
+                    for (validator_address, amount) in stake_per_validator {
+                        // Retrieve the expected validator stake from the committee.
+                        let Some((expected_amount, _)) = committee.members().get(validator_address) else {
+                            bail!(
+                                "Ratify::Genesis(..) found a validator in the bonded balances that is not in the committee"
+                            )
+                        };
+                        // Ensure the staked amount matches the committee.
+                        ensure!(
+                            *expected_amount == amount,
+                            "Ratify::Genesis(..) inconsistent staked amount for validator {validator_address}",
+                        );
                     }
 
                     // Construct the next committee map and next bonded map.
                     let (next_committee_map, next_bonded_map) =
-                        to_next_commitee_map_and_bonded_map(committee, &stakers);
+                        to_next_commitee_map_and_bonded_map(committee, bonded_balances);
 
                     // Insert the next committee into storage.
-                    store.committee_store().insert(state.block_height(), committee.clone())?;
+                    store.committee_store().insert(state.block_height(), *(committee.clone()))?;
                     // Store the finalize operations for updating the committee and bonded mapping.
                     finalize_operations.extend(&[
```

Comment:
With #2213 merged, we need to update the `metadata` mapping here in the Genesis ratification.